### PR TITLE
fixed cucumber-jvm support 

### PR DESCRIPTION
**Pull request by anandbagmar**

### CODE REVIEW

**Comment by saikrishna321** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/com/cucumber/listener/CucumberScenarioReporterListener.java  1
- src/test/java/com/test/cucumber/SampleSteps.java  1
- src/test/java/com/test/cucumber/RunCukes.java  4
         

Complexity decreasing per file
==============================
+ src/test/java/com/appium/executor/MyTestExecutorTest.java  -3
+ src/main/java/com/appium/manager/ATDRunner.java  -2
         

Clones removed
==============
+ src/test/java/com/appium/executor/MyTestExecutorTest.java  -5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/saikrishna321/AppiumTestDistribution/pullRequest?prid=6331726&bid=20572587)

**Comment by saikrishna321** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/com/test/cucumber/SampleSteps.java  1
- src/test/java/com/test/cucumber/RunCukes.java  4
- src/main/java/com/cucumber/listener/CucumberScenarioReporterListener.java  1
         

Complexity decreasing per file
==============================
+ src/main/java/com/appium/manager/ATDRunner.java  -2
+ src/test/java/com/appium/executor/MyTestExecutorTest.java  -3
         

Clones removed
==============
+ src/test/java/com/appium/executor/MyTestExecutorTest.java  -5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/saikrishna321/AppiumTestDistribution/pullRequest?prid=6331726&bid=20572587)

**Comment by saikrishna321** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/com/test/cucumber/RunCukes.java  1
- src/test/java/com/test/cucumber/SampleSteps.java  1
- src/main/java/com/cucumber/listener/CucumberScenarioListener.java  4
- src/main/java/com/cucumber/listener/CucumberScenarioReporterListener.java  1
         

Complexity decreasing per file
==============================
+ src/main/java/com/appium/manager/ATDRunner.java  -2
+ src/test/java/com/appium/executor/MyTestExecutorTest.java  -3
         

Clones removed
==============
+ src/test/java/com/appium/executor/MyTestExecutorTest.java  -5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/saikrishna321/AppiumTestDistribution/pullRequest?prid=6331726&bid=20572587)

**Comment by saikrishna321** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/com/cucumber/listener/CucumberScenarioListener.java  4
- src/test/java/com/test/cucumber/SampleSteps.java  1
- src/test/java/com/test/cucumber/RunCukes.java  1
- src/main/java/com/cucumber/listener/CucumberScenarioReporterListener.java  1
         

Complexity decreasing per file
==============================
+ src/main/java/com/appium/manager/ATDRunner.java  -2
+ src/test/java/com/appium/executor/MyTestExecutorTest.java  -3
         

Clones removed
==============
+ src/test/java/com/appium/executor/MyTestExecutorTest.java  -5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/saikrishna321/AppiumTestDistribution/pullRequest?prid=6331726&bid=20572587)

File: src/test/java/com/test/cucumber/RunCukes.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+                "message:target/results.ndjson",
+                "timeline:target/timline"
+        })
+public class RunCukes extends AbstractTestNGCucumberTests {
+    public RunCukes() {
+        System.out.printf("ThreadID: %d: RunCucumberTest", Thread.currentThread().getId());
+    }
+
+    @Override
+    @DataProvider(parallel = true)
```
Comment (SrinivasanTarget) [inline src/test/java/com/test/cucumber/RunCukes.java:21]: Can this be part of the listener somehow by default rather than users specifying it?

**Review by SrinivasanTarget — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -237,6 +233,14 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
                 <version>2.22.2</version>
+                <configuration>
+                    <properties>
+                        <property>
+                            <name>dataproviderthreadcount</name>
+                            <value>1</value>
```
Comment (SrinivasanTarget) [inline pom.xml:240]: Let's try increasing this and run on both iOS and Android

Comment (anandbagmar) [reply pom.xml:240]: @SrinivasanTarget - I tested this with 2 android devices. Do not have an iOS device to try that.

File: src/test/java/com/test/cucumber/RunCukes.java (hunk 1 above)
Comment (anandbagmar) [reply src/test/java/com/test/cucumber/RunCukes.java:21]: Need to see if that can be done. If yes, ofcourse.

File: pom.xml (hunk 2 above)
Comment (SrinivasanTarget) [reply pom.xml:240]: I will try this in iOS Simulators and keep you posted.

**Comment by saikrishna321** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/com/cucumber/listener/CucumberScenarioReporterListener.java  1
- src/test/java/com/test/cucumber/RunCukes.java  1
- src/main/java/com/cucumber/listener/CucumberScenarioListener.java  4
- src/test/java/com/test/cucumber/SampleSteps.java  1
         

Complexity decreasing per file
==============================
+ src/test/java/com/appium/executor/MyTestExecutorTest.java  -3
+ src/main/java/com/appium/manager/ATDRunner.java  -2
         

Clones removed
==============
+ src/test/java/com/appium/executor/MyTestExecutorTest.java  -5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/saikrishna321/AppiumTestDistribution/pullRequest?prid=6331726&bid=20572587)

**Comment by saikrishna321** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/com/test/cucumber/SampleSteps.java  1
- src/main/java/com/cucumber/listener/CucumberScenarioReporterListener.java  1
- src/main/java/com/cucumber/listener/CucumberScenarioListener.java  4
- src/test/java/com/test/cucumber/RunCukes.java  1
         

Complexity decreasing per file
==============================
+ src/main/java/com/appium/manager/ATDRunner.java  -2
+ src/test/java/com/appium/executor/MyTestExecutorTest.java  -3
         

Clones removed
==============
+ src/test/java/com/appium/executor/MyTestExecutorTest.java  -5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/saikrishna321/AppiumTestDistribution/pullRequest?prid=6331726&bid=20572587)

**Pull request merged**